### PR TITLE
[FIRRTL] Block dedup on input RefType ports

### PR DESCRIPTION
**Pull request by seldridge**
Change FIRRTL's structural deduplication ("dedup") pass to block dedup if a module has input RefType ports.  This is done because input RefType ports are consumed by a RefResolveOp which will lower to a Verilog hierarchical reference (XMR).  If dedup is allowed, this XMR may not have a valid representation because it refers to two different things in the circuit.

Signed-off-by: Schuyler Eldridge <schuyler.eldridge@sifive.com>

### CODE REVIEW

**Review by prithayan — APPROVED** [review]
Looks good. But maybe this should be merged last, after the other PRs are merged and after getting a +1 from the unit level verification flow.

**Comment by prithayan** [comment]
Ah, I just noticed everything is getting merged into the same branch. Neat.

**Review by darthscsi — APPROVED** [review]

**Comment by seldridge** [comment]
Merged via: https://github.com/llvm/circt/commit/acf3e44210dff56a3930e87dc6d1cbdf5c65b4f6

**Pull request closed without merging**
